### ISSUE TITLE
[BUG] Multithreaded Crash with SKShaper and SKTypeFace

### ISSUE THREAD

**Issue by mlptownsend** [mono/SkiaSharp#1220]
**Description**

Well, it may not necessarily be SKShaper related, but it's how I can make it happen. I can reproduce it by adding the below unit test code to the SKShaperTest class. Importantly, the typeface class has completely lost its Handle value! It'll do it whether I wrap the typeface in a using block or not. I think something is prematurely disposing and eliminating typefaces that might still be in play? Sometimes the unit test will complete, so be patient and try it a few times. In debug mode I had to remove the THROW_OBJECT_EXCEPTIONS compile flag to catch it.

It didn't happen with 1.68.0, but it's definitely happening with 1.68.1.1.

Same code will reproduce it via a .NET Core Console app too. Started with that but figured I'd get a little more info with the extra debug info from the test project instead of it killing the process outright.

**Code**

```
[SkippableFact]
		public void TestMultiThreadCrash()
		{
			const int count = 20000;
			var threads = new List<Thread>();

			for (var i = 0; i < count; i++)
			{
				var t = new Thread(new ParameterizedThreadStart(Work));
				t.TrySetApartmentState(Thread.CurrentThread.GetApartmentState());
				t.Name = i.ToString();
				t.Start(i);
				threads.Add(t);
			}
			for (var i = 0; i < count; i++)
			{
				threads[i].Join();
			}
		}
		private void Work(object state)
		{
			var text = $"Some wonderful text goes here {state}.";
			var typeface = SkiaSharp.SKTypeface.FromFamilyName("Arial", SkiaSharp.SKFontStyle.Bold);
			//using (var typeface = SkiaSharp.SKTypeface.FromFamilyName("Arial", SkiaSharp.SKFontStyle.Bold))
			//{
			SkiaSharp.HarfBuzz.SKShaper.Result result;

			using (var paint = new SkiaSharp.SKPaint())
			{

				using (var shaper = new SkiaSharp.HarfBuzz.SKShaper(typeface))
				{
					result = shaper.Shape(text, 75, 75, paint);
				}

				var imageInfo = new SkiaSharp.SKImageInfo(128, 128);
				using (var bitmap = new SkiaSharp.SKBitmap(imageInfo))
				using (var canvas = new SkiaSharp.SKCanvas(bitmap))
				{
					var bytes = result.Codepoints.Select(cp => BitConverter.GetBytes((ushort)cp)).SelectMany(b => b).ToArray();
					paint.TextEncoding = SkiaSharp.SKTextEncoding.GlyphId;
					canvas.DrawPositionedText(bytes, result.Points, paint);
				}
			}
			//}
		}
```

As console app:

```
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading;

namespace ConsoleApp {
	class Program {

		static void Main(string[] args) {

			Console.WriteLine($"IntPtr size: {IntPtr.Size}");

			const int count = 100;
			var threads = new List<Thread>();

			for (var i = 0; i < count; i++) {
				var t = new Thread(new ParameterizedThreadStart(Work));
				t.TrySetApartmentState(ApartmentState.MTA);
				t.Name = i.ToString();
				t.Start(i);
				threads.Add(t);
			}
			for (var i = 0; i < count; i++) {
				threads[i].Join();
			}
			Console.WriteLine("Done.");
		}

		private static void Work(object state) {
			var text = $"Some wonderful text goes here {state}.";
			var typeface = SkiaSharp.SKTypeface.FromFamilyName("Arial", SkiaSharp.SKFontStyle.Bold);
			SkiaSharp.HarfBuzz.SKShaper.Result result;

			using var paint = new SkiaSharp.SKPaint();

			using (var shaper = new SkiaSharp.HarfBuzz.SKShaper(typeface)) {
				result = shaper.Shape(text, 75, 75, paint);
			}

			var imageInfo = new SkiaSharp.SKImageInfo(128, 128);
			using var bitmap = new SkiaSharp.SKBitmap(imageInfo);
			using var canvas = new SkiaSharp.SKCanvas(bitmap);

			var bytes = result.Codepoints.Select(cp => BitConverter.GetBytes((ushort)cp)).SelectMany(b => b).ToArray();
			paint.TextEncoding = SkiaSharp.SKTextEncoding.GlyphId;
			canvas.DrawPositionedText(bytes, result.Points, paint);
		}

	}
}
```

**Expected Behavior**

Typefaces shouldn't be disappearing?

**Actual Behavior**

Typefaces are disappearing!

**Basic Information**

- Version with issue:  1.68.1.1
- Last known good version:  1.68.0
- IDE:  VS2017/2019
- Platform Target Frameworks: 
  - Windows Classic:  Win10 via xUnit x86 / .NET Core 3 x64

**Screenshots**

![DisappearingHandle](https://user-images.githubusercontent.com/38544371/78842088-b676a080-79c4-11ea-9dc0-20924906506a.png)

![TypefaceCrashSideBySide](https://user-images.githubusercontent.com/38544371/78842395-77951a80-79c5-11ea-9d15-e3746b26e928.png)

**Comment by mattleibow** [comment]
Thanks for this issue, it brought up an interesting case where sometimes the font manager returns the same object. This results in the disposal of one causing issues for the rest.

I have a fix that I am testing locally that should correctly identify and prevent this from happening.

**Comment by mlptownsend** [comment]
Awesome, thanks!

**Comment by mattleibow** [comment]
Released to NuGet as version **1.68.2-preview.60**
